api/envoy/extensions/filters/network/mysql_proxy/v3/mysql_proxy.proto

```diff
@@ -15,15 +21,36 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // MySQL Proxy :ref:`configuration overview <config_network_filters_mysql_proxy>`.
 // [#extension: envoy.filters.network.mysql_proxy]
 
+// [#next-free-field: 6]
 message MySQLProxy {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.network.mysql_proxy.v1alpha1.MySQLProxy";
 
+  message Route {
+    // String database name that must exact match with client connection db name
+    string database = 1;
+
+    // Upstream cluster to forward the command to.
+    string cluster = 2;
+  }
+
   // The human readable prefix to use when emitting :ref:`statistics
   // <config_network_filters_mysql_proxy_stats>`.
   string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
 
   // [#not-implemented-hide:] The optional path to use for writing MySQL access logs.
   // If the access log field is empty, access logs will not be written.
   string access_log = 2;
+
+  repeated Route routes = 3;
```

Comment:
Here I will add a field like `bool new_feature_enabled` to control after I replace the signature of `isTerminalFilter`. https://github.com/envoyproxy/envoy/issues/15565#issuecomment-805835760